apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/shard/helper/CsvImporterImpl.java

```diff
@@ -172,53 +174,27 @@ private long importCsv(String tableName, int batchLimit, boolean cleanTarget,
+                        } catch (SQLException e) {
                             log.error("Binary/Varbinary reading error = " + object, e);
                             throw e;
                         }
                         // ignore error here
                     } else if (object != null && (meta.getColumnType(i + 1) == Types.ARRAY)) {
-                        String objectArray = (String)object;
-                        Object[] actualArray;
-                        if (!StringUtils.isBlank(objectArray)) {
-                            String[] split = objectArray.split(",");
-                            actualArray = new Object[split.length];
-                            for (int j = 0; j < split.length; j++) {
-                                String value = split[j];
-                                if (value.startsWith("b\'") && value.endsWith("\'")) { //find byte arrays
-                                    //byte array found
-                                    byte[] actualValue = Base64.getDecoder().decode(value.substring(2, value.length() - 1));
-                                    actualArray[j] = actualValue;
-                                } else if (value.startsWith("\'") && value.endsWith("\'")) { //find string
-                                    actualArray[j] = split[j].substring(1, split[j].length() - 1);
-                                } else { // try to process number
-                                    try {
-                                        actualArray[j] = Integer.parseInt(split[j]);
-                                    }
-                                    catch (NumberFormatException ignored) { // value can be of long type
-                                        try {
-                                            actualArray[j] = Long.parseLong(split[j]); // try to parse long
-                                        }
-                                        catch (NumberFormatException e) { // throw exception, when specified value is not a string, long, int or byte array
-                                            throw new RuntimeException("Value " + split[j] + " of unsupported type");
-                                        }
-                                    }
-                                }
-                            }
-                        } else {
-                            actualArray = new Object[0];
-                        }
+                        Object[] actualArray = parser.parseArrayObject(object);
                         SimpleResultSet.SimpleArray simpleArray = new SimpleResultSet.SimpleArray(actualArray);
                         preparedInsertStatement.setArray(i + 1, simpleArray);
                         row.put(columnName.toLowerCase(), actualArray);
+                    } else if (object != null && (meta.getColumnType(i + 1) == Types.VARCHAR || meta.getColumnType(i + 1) == Types.NVARCHAR)) {
```

Comment:
I suggest putting this condition in a separate method with an implicit name. For example:
isVarcharColumnTypeAndNonNull...

As a result, a further maintainer might understand it fast

Reply:
done